public/components/Flyout/flyouts/components/AlertsDashboardFlyoutComponent.test.js

```diff
@@ -14,6 +15,8 @@ describe('AlertsDashboardFlyoutComponent', () => {
         location={{ pathname: '/dashboard', search: '' }}
         flyout={{ type: 'message', payload: null }}
         onClose={jest.fn()}
+        history={historyMock}
```

Comment:
Why was the test failing though?

Reply:
I noticed the test failed in PR https://github.com/opensearch-project/alerting-dashboards-plugin/pull/574 with the following error. Adding the mock resolved the undefined issue. After resolving that issue, the test failed with the error mentioned in [the comment above](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/576#discussion_r1231624934).
```
/home/runner/work/alerting-dashboards-plugin/alerting-dashboards-plugin/OpenSearch-Dashboards/plugins/alerting-dashboards-plugin/public/components/Flyout/flyouts/components/AlertsDashboardFlyoutComponent.js:4357
              history.replace(
                      ^

TypeError: Cannot read properties of undefined (reading 'replace')
```
https://github.com/opensearch-project/alerting-dashboards-plugin/actions/runs/5273730162/jobs/9537784657?pr=574#step:11:422